src/Java.Interop.Tools.JavaCallableWrappers/Java.Interop.Tools.JavaCallableWrappers/CRC64.cs

```diff
@@ -0,0 +1,156 @@
+		    0xe8a46c1224f5a634, 0x927c1cda14c02f4d,
+		    0x1d148d82449eb4c6, 0x67ccfd4a74ab3dbf,
+		    0x289c8961bcb410bb, 0x5244f9a98c8199c2,
+		    0xdd2c68f1dcdf0249, 0xa7f41839ecea8b30,
+		    0x438c80a64ce15841, 0x3954f06e7cd4d138,
+		    0xb63c61362c8a4ab3, 0xcce411fe1cbfc3ca,
+		    0x83b465d5d4a0eece, 0xf96c151de49567b7,
+		    0x76048445b4cbfc3c, 0x0cdcf48d84fe7545,
+		    0x6fbd6d5ebd3716b7, 0x15651d968d029fce,
+		    0x9a0d8ccedd5c0445, 0xe0d5fc06ed698d3c,
+		    0xaf85882d2576a038, 0xd55df8e515432941,
+		    0x5a3569bd451db2ca, 0x20ed197575283bb3,
+		    0xc49581ead523e8c2, 0xbe4df122e51661bb,
+		    0x3125607ab548fa30, 0x4bfd10b2857d7349,
+		    0x04ad64994d625e4d, 0x7e7514517d57d734,
+		    0xf11d85092d094cbf, 0x8bc5f5c11d3cc5c6,
+		    0x12b5926535897936, 0x686de2ad05bcf04f,
+		    0xe70573f555e26bc4, 0x9ddd033d65d7e2bd,
+		    0xd28d7716adc8cfb9, 0xa85507de9dfd46c0,
+		    0x273d9686cda3dd4b, 0x5de5e64efd965432,
+		    0xb99d7ed15d9d8743, 0xc3450e196da80e3a,
+		    0x4c2d9f413df695b1, 0x36f5ef890dc31cc8,
+		    0x79a59ba2c5dc31cc, 0x037deb6af5e9b8b5,
+		    0x8c157a32a5b7233e, 0xf6cd0afa9582aa47,
+		    0x4ad64994d625e4da, 0x300e395ce6106da3,
+		    0xbf66a804b64ef628, 0xc5bed8cc867b7f51,
+		    0x8aeeace74e645255, 0xf036dc2f7e51db2c,
+		    0x7f5e4d772e0f40a7, 0x05863dbf1e3ac9de,
+		    0xe1fea520be311aaf, 0x9b26d5e88e0493d6,
+		    0x144e44b0de5a085d, 0x6e963478ee6f8124,
+		    0x21c640532670ac20, 0x5b1e309b16452559,
+		    0xd476a1c3461bbed2, 0xaeaed10b762e37ab,
+		    0x37deb6af5e9b8b5b, 0x4d06c6676eae0222,
+		    0xc26e573f3ef099a9, 0xb8b627f70ec510d0,
+		    0xf7e653dcc6da3dd4, 0x8d3e2314f6efb4ad,
+		    0x0256b24ca6b12f26, 0x788ec2849684a65f,
+		    0x9cf65a1b368f752e, 0xe62e2ad306bafc57,
+		    0x6946bb8b56e467dc, 0x139ecb4366d1eea5,
+		    0x5ccebf68aecec3a1, 0x2616cfa09efb4ad8,
+		    0xa97e5ef8cea5d153, 0xd3a62e30fe90582a,
+		    0xb0c7b7e3c7593bd8, 0xca1fc72bf76cb2a1,
+		    0x45775673a732292a, 0x3faf26bb9707a053,
+		    0x70ff52905f188d57, 0x0a2722586f2d042e,
+		    0x854fb3003f739fa5, 0xff97c3c80f4616dc,
+		    0x1bef5b57af4dc5ad, 0x61372b9f9f784cd4,
+		    0xee5fbac7cf26d75f, 0x9487ca0fff135e26,
+		    0xdbd7be24370c7322, 0xa10fceec0739fa5b,
+		    0x2e675fb4576761d0, 0x54bf2f7c6752e8a9,
+		    0xcdcf48d84fe75459, 0xb71738107fd2dd20,
+		    0x387fa9482f8c46ab, 0x42a7d9801fb9cfd2,
+		    0x0df7adabd7a6e2d6, 0x772fdd63e7936baf,
+		    0xf8474c3bb7cdf024, 0x829f3cf387f8795d,
+		    0x66e7a46c27f3aa2c, 0x1c3fd4a417c62355,
+		    0x935745fc4798b8de, 0xe98f353477ad31a7,
+		    0xa6df411fbfb21ca3, 0xdc0731d78f8795da,
+		    0x536fa08fdfd90e51, 0x29b7d047efec8728,
+		};
+
+		ulong crc;
+
+		public override void Initialize () { }
+
+		protected override void HashCore (byte [] array, int ibStart, int cbSize)
+		{
+			for (int i = ibStart; i < cbSize; i++) {
+				crc = Table [(byte) (crc ^ array [i])] ^ (crc >> 8);
+			}
+		}
+
+		protected override byte [] HashFinal () => BitConverter.GetBytes (crc);
```

Comment:
Is "endianness" a problem here? Would I get different results on different systems?

Reply:
It would be a problem if you sent it across the network to a different machine in which case the receiver might get the same data but in a different order, thus resulting in a different value. It shouldn't be a problem otherwise

Reply:
Well, someone might put `crc1234.MainActivity` in their manifest and building on another machine produces `crc4321.MainActivity` and the manifest no longer matches?

Reply:
The calculation would yield the same result - the machine would be working in its "native" byte order, so no conversion would be necessary (in other words, `0x0502 | 0x0201` will yield the same result of `0x703` even though the byte order of the 16-bit word will be different on little- and bigendian machines).